File: FacebookSDK.Tests/MockHttpClient.cpp
#include "pch.h"
#include "MockHttpClient.h"
#include <ppltasks.h>

using namespace Platform;
using namespace concurrency;
using namespace Windows::Foundation;
using namespace Windows::Foundation::Collections;

namespace FacebookSDK_Tests
{

	IAsyncOperation<String^>^ MockHttpClient::GetTaskAsync(String^ path, IMapView<String^, Object^>^ parameters)
	{
		return create_async([=]() {
			return ResponseData;
		});
	}

	Windows::Foundation::IAsyncOperation<Platform::String^>^ MockHttpClient::PostTaskAsync(
		Platform::String^ path,
		Windows::Foundation::Collections::IMapView<Platform::String^, Platform::Object^>^ parameters
	) {
		throw ref new Platform::NotImplementedException();
	}

	Windows::Foundation::IAsyncOperation<Platform::String^>^ MockHttpClient::DeleteTaskAsync(
		Platform::String^ path,
		Windows::Foundation::Collections::IMapView<Platform::String^, Platform::Object^>^ parameters
	) {
		throw ref new Platform::NotImplementedException();
	}

	Platform::String^ MockHttpClient::ParametersToQueryString(
		Windows::Foundation::Collections::IMapView<Platform::String^, Platform::Object^>^ parameters
	) {
		throw ref new Platform::NotImplementedException();
	}
}

File: winsdkfb.CS.Tests/MockHttpClient.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Windows.Foundation;
using winsdkfb;

namespace winsdkfb.CS.Tests
{
    class MockHttpClient : IHttpClient
    {
        public String ResponseData;

        public IAsyncOperation<string> DeleteTaskAsync(string path, IReadOnlyDictionary<String, Object> parameters)
        {
            return Task.Run(() =>
            {
                return ResponseData;
            }).AsAsyncOperation<string>();
        }

        public IAsyncOperation<string> GetTaskAsync(string path, IReadOnlyDictionary<String, Object> parameters)
        {
            return Task.Run(() =>
            {
                return ResponseData;
            }).AsAsyncOperation<string>();
        }

        public string ParametersToQueryString(IReadOnlyDictionary<String, Object> Parameters)
        {
            throw new NotImplementedException();
        }

        public IAsyncOperation<string> PostTaskAsync(string path, IReadOnlyDictionary<String, Object> parameters)
        {
            return Task.Run(() =>
            {
                return ResponseData;
            }).AsAsyncOperation<string>();
        }
    }
}
